sklearn/preprocessing/tests/test_polynomial.py

```diff
@@ -552,6 +552,33 @@ def test_polynomial_features_csr_X(deg, include_bias, interaction_only, dtype):
     assert_array_almost_equal(Xt_csr.A, Xt_dense)
 
 
+@pytest.mark.parametrize("columns", [1, 2, 3, 1000])
+def test_polynomial_features_csr_wide(columns):
+    x = sparse.csr_matrix(([1], ([0], [columns])))
+
+    est = PolynomialFeatures(3)
+    # this causes a crash with values of `columns` which are too large
+    est.fit_transform(x)
+
+
+@pytest.mark.parametrize("n_features", range(1, 5))
```

Comment:
The parametrization generates 100 tests. Would it be enough to have `"n_features", [1, 4, 5]`?

Reply:
This actually takes a similar amount of time to run because we exchange 2,3 for 5 and the old algorithm slows down quickly with increased degree:

```
====================================================================================== test session starts =======================================================================================
platform linux -- Python 3.9.2, pytest-6.2.2, py-1.10.0, pluggy-0.13.1
rootdir: /home/frederick/Projects/scikit-learn, configfile: setup.cfg
plugins: cov-2.11.1
collected 203 items / 139 deselected / 64 selected                                                                                                                                               

tests/test_polynomial.py ................................................................                                                                                                  [100%]

=============================================================================== 64 passed, 139 deselected in 0.37s ===============================================================================
```
vs
```(sklearn-venv) frederick@penguin ~/Projects/scikit-learn/sklearn/preprocessing % pytest tests/test_polynomial.py -k test_num_combinations
====================================================================================== test session starts =======================================================================================
platform linux -- Python 3.9.2, pytest-6.2.2, py-1.10.0, pluggy-0.13.1
rootdir: /home/frederick/Projects/scikit-learn, configfile: setup.cfg
plugins: cov-2.11.1
collected 187 items / 139 deselected / 48 selected                                                                                                                                               

tests/test_polynomial.py ................................................                                                                                                                  [100%]

=============================================================================== 48 passed, 139 deselected in 0.32s ===============================================================================
```